clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreWriteComputeProcessor.java

```diff
@@ -118,12 +120,34 @@ private SchemaAndUniqueId getSchemaAndUniqueId(int valueSchemaId, int writeCompu
     return schemaAndUniqueIdCache.get(valueSchemaId, writeComputeSchemaId);
   }
 
-  private AvroSerializer<GenericRecord> getValueSerializer(int valueSchemaId) {
+  RecordDeserializer<GenericRecord> getValueDeserializer(Schema writerSchema, Schema readerSchema) {
+    if (fastAvroEnabled) {
+      return FastSerializerDeserializerFactory.getFastAvroGenericDeserializer(writerSchema, readerSchema);
+    }
+
+    /**
+     * TODO: why do we need to preserve the map-order since this class is only used by {@link LeaderFollowerStoreIngestionTask}.
+     * We are not supposed to run DCR validation workflow against non-AA store.
+     */
+    // Map in write compute needs to have consistent ordering. On the sender side, users may not care about ordering
+    // in their maps. However, on the receiver side, we still want to make sure that the same serialized map bytes
+    // always get deserialized into maps with the same entry ordering.
+    return MapOrderingPreservingSerDeFactory.getDeserializer(writerSchema, readerSchema);
+  }
+
+  private RecordSerializer<GenericRecord> getValueSerializer(int valueSchemaId) {
     return serializerIndexedBySchemaId.computeIfAbsent(valueSchemaId, this::generateValueSerializer);
   }
 
-  private AvroSerializer<GenericRecord> generateValueSerializer(int valueSchemaId) {
+  RecordSerializer<GenericRecord> generateValueSerializer(int valueSchemaId) {
     Schema valueSchema = getValueSchema(valueSchemaId);
+    /**
+     * TODO: why do we need to preserve the map-order since this class is only used by {@link LeaderFollowerStoreIngestionTask}.
```

Comment:
So you are certain that this class is not used in AA, right?

Reply:
The only public API "applyWriteCompute" is only used by LFSIT#handleUpdateRequest, which should only be used by non-AA store.

Reply:
yeah, I am certain based on the call stack, and AA is using this method:
https://github.com/linkedin/venice/blob/main/internal/venice-client-common/src/main/java/com/linkedin/venice/schema/writecompute/WriteComputeProcessor.java#L58

Reply:
Ok I see. It makes sense that LF and AA have separate code paths, since their logic needs to be a bit different. On the other hand, it seems a bit hard to maintain, and the class names are very similar (of course, since they do almost the same thing...). Hopefully, we could find a better way to structure this code in the future.